### PR TITLE
[Testing] Test VM creation via command line

### PR DESCRIPTION
**Pull request by stu-gott**
```release-note
NONE
```

### CODE REVIEW

**Comment by ILpinto** [comment]
LGTM we can merge it

File: tests/vm_test.go [hunk 1]
```diff
@@ -1018,9 +1026,184 @@ var _ = Describe("[rfe_id:1177][crit:medium][vendor:cnv-qe@redhat.com][level:com
...
+			}, 60*time.Second, 1*time.Second).Should(Equal(vm.Name), "New VMI was not created")
+
+			By("Listing running pods")
+			stdout, _, err := tests.RunCommand(k8sClient, "get", "pods")
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Ensuring pod is running")
+			found := false
+			expectedPodName := getExpectedPodName(vm)
+			for _, line := range strings.Split(strings.TrimSuffix(stdout, "\n"), "\n") {
```
Comment (davidvossel) [inline tests/vm_test.go]: regex here?

File: tests/vm_test.go [hunk 2]
```diff
@@ -1018,9 +1026,184 @@ var _ = Describe("[rfe_id:1177][crit:medium][vendor:cnv-qe@redhat.com][level:com
...
+			_, _, err := tests.RunCommand(k8sClient, "create", "-f", vmJson)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Invoking virtctl start")
+			virtctl := tests.NewRepeatableVirtctlCommand(vm.COMMAND_START, "--namespace", thisVm.Namespace, thisVm.Name)
+			err = virtctl()
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Waiting for VMI to start")
+			Eventually(func() string {
```
Comment (davidvossel) [inline tests/vm_test.go]: good candidate for code reuse function

File: tests/vm_test.go [hunk 3]
```diff
@@ -1018,9 +1026,184 @@ var _ = Describe("[rfe_id:1177][crit:medium][vendor:cnv-qe@redhat.com][level:com
...
+					return newVMI.Name
+				}
+				return ""
+			}, 120*time.Second, 1*time.Second).Should(Equal(thisVm.Name), "New VMI was not created")
+
+			By("Checking that VM is running")
+			stdout, _, err := tests.RunCommand(k8sClient, "describe", "vmis", thisVm.GetName())
+			Expect(err).ToNot(HaveOccurred())
+			found := false
+			for _, line := range strings.Split(strings.TrimSuffix(stdout, "\n"), "\n") {
```
Comment (davidvossel) [inline tests/vm_test.go]: candidate for helper function

File: tests/vm_test.go [hunk 4]
```diff
@@ -1018,9 +1026,184 @@ var _ = Describe("[rfe_id:1177][crit:medium][vendor:cnv-qe@redhat.com][level:com
...
+
+			vmJson, err = tests.GenerateVMJson(thisVm, workDir)
+			Expect(err).ToNot(HaveOccurred(), "Cannot generate VM's manifest")
+
+			By("Creating VM using k8s client binary")
+			_, _, err := tests.RunCommand(k8sClient, "create", "-f", vmJson)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Waiting for VMI to start")
+			Eventually(func() v1.VirtualMachineInstancePhase {
```
Comment (davidvossel) [inline tests/vm_test.go]: help function

File: tests/vm_test.go [hunk 5]
```diff
@@ -1018,9 +1026,184 @@ var _ = Describe("[rfe_id:1177][crit:medium][vendor:cnv-qe@redhat.com][level:com
...
+				Expect(err).ToNot(HaveOccurred())
+				return strings.Contains(stdout, thisVm.GetName())
+			}, 120*time.Second, 1*time.Second).Should(BeFalse(), "VM was not deleted")
+
+			By("Creating same VM using k8s client binary and same manifest")
+			_, _, err = tests.RunCommand(k8sClient, "create", "-f", vmJson)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Waiting for VMI to start")
+			Eventually(func() v1.VirtualMachineInstancePhase {
```
Comment (davidvossel) [inline tests/vm_test.go]: helper function

File: tests/datavolume_test.go [hunk 6]
```diff
@@ -155,8 +157,18 @@ var _ = Describe("DataVolume Integration", func() {
...
 			pvcName = dataVolumeName
 
-			vmJson, err = tests.GenerateVMJson(vm)
+			workDir, err := ioutil.TempDir("", tests.TempDirPrefix+"-")
 			Expect(err).ToNot(HaveOccurred())
+			vmJson, err = tests.GenerateVMJson(vm, workDir)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		AfterEach(func() {
```
[the remark this answers is not in the document]
Comment (stu-gott) [reply tests/datavolume_test.go]: oops. this is a bug introduced by rebase. will fix

**Comment by stu-gott** [comment]
@davidvossel Code squished.

**Pull request merged**